docs/development/testguide.rst

```diff
@@ -117,6 +100,27 @@ This can be used in conjunction with the ``-P`` option provided by the
 `pytest-filter-subpackage <https://github.com/astropy/pytest-filter-subpackage>`_
 plugin to run just part of the test suite.
 
+Using tox to set up a development environment
+---------------------------------------------
+
+As noted, running ``tox`` can be quite slow, yet it has the advantage over the
+other methods that it puts together a consistent set of dependencies.  One can
+speed up matters by using ``tox`` to set up a virtual environment for
+development and then use direct ``pytest`` commands within that environment.
+For instance::
+
+    tox -e test-alldeps --develop --notest
+    source .tox/test-alldeps/bin/activate
```

Comment:
Okay, I did find one thing that needs clarification. I did the `source` thingy and now I am inside a `tox` environment and I don't know how to get out of it. It doesn't come with `deactivate`.

Reply:
Running `deactivate` should work (it's a shell function defined in the `activate` script).

Reply:
Do this work with `--develop` ? As discussed in https://github.com/astropy/astropy/issues/10303#issuecomment-623787982 currently `--develop` is not supported by PEP 517 / 518, so the dependencies in `pyproject.toml` are not installed.
Also another way to proceed without activating the env would be to run directly `.tox/test-alldeps/bin/pytest`.

Reply:
@pllim - I also found that `deactivate` thing confusing for exactly the same reason as you. I had to google to just convince myself to try typing `deactivate`.  Which means that documenting it hear might be worthwhile.

Reply:
Added `deactivate` to the description.